consensus/src/lib.rs

```diff
@@ -50,11 +68,33 @@ pub enum ConsensusCreationError {
     MiningStopped,
     #[error("Mining failed")]
     MiningFailed,
+    #[error("Staking error")]
+    StakingError(#[from] ConsensusPoSError),
+    #[error("Staking failed")]
+    StakingFailed,
+    #[error("Staking stopped")]
+    StakingStopped,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum GenerateBlockInputData {
+    #[codec(index = 0)]
+    PoW,
+    #[codec(index = 1)]
+    PoS(Box<PoSGenerateBlockInputData>),
+}
+
+#[derive(Debug, Clone, Encode, Decode)]
+pub enum FinalizeBlockInputData {
+    PoW,
+    PoS(PoSFinalizeBlockInputData),
 }
 
 pub fn generate_consensus_data<G>(
     chain_config: &ChainConfig,
     prev_block_index: &GenBlockIndex,
+    sealed_epoch_randomness: Option<EpochData>,
```

Comment:
can randomness be part of InputData where all consensus specific data is stored? It's slightly confusing to see it in this generic function

Reply:
my concern is that randomness is only valid for PoS and should be somehow encapsulated. Maybe we can get back to this later I won't insist. 

Btw name is randomness but the type is EpochData.